### PR TITLE
Reuse SQL result for module history

### PR DESCRIPTION
**Pull request by Quetzacoalt91**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | Instead of executing the same sql request more than 50 times, reuse the results. |
| Type? | improvement |
| Category? | BO |
| BC breaks? | Nope |
| Deprecations? | Nope |
| Fixed ticket? | / |
| How to test? | Look at the debug bar at the bottom of the module page > Installed module tab |

Before:
capture d ecran 2016-10-13 a 22 46 49

After:
capture d ecran 2016-10-13 a 22 46 19

### CODE REVIEW

File: src/Adapter/Module/ModuleDataProvider.php [hunk 1]
```diff
@@ -71,25 +71,26 @@ public function findByName($name)
...
+                
+                $qb = $this->entityManager->createQueryBuilder();
+                $qb->select('mh')
+                    ->from('PrestaShopBundle:ModuleHistory', 'mh', 'mh.idModule')
+                    ->where('mh.idEmployee = ?1')
+                    ->setParameter(1, $employeeID);
+                $query = $qb->getQuery();
+                $query->useResultCache(true);
+                $modulesHistory = $query->getResult();
+dump($modulesHistory);
```
Comment (thierrymarianne) [inline src/Adapter/Module/ModuleDataProvider.php]: oops

Comment (Quetzacoalt91) [inline src/Adapter/Module/ModuleDataProvider.php]: Removed

**Comment by aleeks** [comment]
Thank you @Quetzacoalt91 !

**Comment by jonathanparentlevesque** [comment]
After updating my table name prefix and `config.php`. My queries seems fine pretty much everywhere, but I keep recieving this error when accessing the module page:

`Unexpected exception on module dashactivity. An exception occurred while executing \'SELECT v0_.id AS id_0, v0_.id_employee AS id_employee_1, v0_.id_module AS id_module_2, v0_.date_add AS date_add_3, v0_.date_upd AS date_upd_4 FROM v2v_psmodule_history v0_ WHERE v0_.id_employee = ?\' with params [1]:\n\nSQLSTATE[42S02]: Base table or view not found: 1146 Table \'woot.v2v_psmodule_history\' doesn\'t exist`

```
PHP Fatal error: Uncaught PDOException: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'woot.v2v_psmodule_history' doesn't exist in C:\xampp\htdocs\woot\vendor\doctrine\dbal\lib\Doctrine\DBAL\Driver\PDOStatement.php:91 
Stack trace: 
#0 C:\xampp\htdocs\woot\vendor\doctrine\dbal\lib\Doctrine\DBAL\Driver\PDOStatement.php(91): PDOStatement->execute(NULL) 
#1 C:\xampp\htdocs\woot\vendor\doctrine\dbal\lib\Doctrine\DBAL\Connection.php(847): Doctrine\DBAL\Driver\PDOStatement->execute() 
#2 C:\xampp\htdocs\woot\vendor\doctrine\dbal\lib\Doctrine\DBAL\Connection.php(899): Doctrine\DBAL\Connection->executeQuery('SELECT v0_.id A...', Array, Array) 
#3 C:\xampp\htdocs\woot\vendor\doctrine\dbal\lib\Doctrine\DBAL\Connection.php(830): Doctrine\DBAL\Connection->executeCacheQuery('SELECT v0_.id A...', Array, Array, Object(Doctrine\DBAL\Cache\QueryCacheProfile)) 
#4 C:\xampp\htdocs\woot\vendor\doctrine\orm\lib\Doctrine\ORM\Query\Exec\SingleSelectExecutor.php(50): Doctrine\DBAL\Connection->executeQuery('SELECT v0_.id A...', Array, Array, Ob in C:\xampp\htdocs\woot\app\bootstrap.php.cache on line 3216
```

**Pull request merged**